Add profile property

michaeltravisuk opened the pull request:
Add profile option for allowing authentication with AWS Shared Credentials file with a named profile.

Allows authenticating with multiple AWS accounts by the same Logstash process.

karmi commented:
Hi @michaeltravisuk, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-input-kinesis/pull/26.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

michaeltravisuk commented:
I've checked my github email address and it matches up to the CLA document I just had emailed to me.  I've made it so that the verified email address shows up on my profile, hopefully this will show up for you.

michaeltravisuk commented:
I've added the email I was using to create the commit, you should be able to see that on my profile.

houseofback commented:
This feature will be a great addition 👍

jordansissel commented:
Can you update the docs for this change? https://github.com/logstash-plugins/logstash-input-kinesis/blob/master/docs/index.asciidoc

jordansissel commented on lib/logstash/inputs/kinesis.rb:
```diff
@@ -50,6 +50,9 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
   config :metrics, :validate => [nil, "cloudwatch"], :default => nil
 
+  # Select AWS profile for input
+  config :profile, :validate => :string, :default => ""
```
I recommend not setting a default, here. Then, below, instead of @profile.empty? you can do @profile.nil?

jordansissel commented on lib/logstash/inputs/kinesis.rb:
```diff
@@ -64,7 +67,14 @@ def register
+    # If the AWS profile is set, use the profile credentials provider.
+    # Otherwise fall back to the default chain.
+    unless @profile.empty?
+      creds = com.amazonaws.auth.profile::ProfileCredentialsProvider.new(String @profile)
```
`String @profile` seems like invalid ruby code?

michaeltravisuk commented:
Thanks @jordansissel for your feedback.

I've updated the code as per your suggestions and added tests to verify the credentials provider being used in each case.

The tests all work locally, but as CI builds are failing against the master branch, I suspect the same will be the case here.

codekitchen commented:
@ph it looks like your changes to the travis build have never passed successfully, are you planning further changes?

codekitchen commented:
CI is fixed now, a rebase on master should get you passing.

michaeltravisuk commented:
Updated to pull in fixed CI code, new tests included to test for newly added functionality, does this look OK to you @jordansissel?

michaeltravisuk commented:
Any update on whether this feature is OK to be merged?

famuyiwad commented:
Just wondering, @michaeltravisuk  @jordansissel. Is there anything else that is stopping this PR from being merged. Thanks

codekitchen commented:
It looks like you've addressed the comments and the rest of this looks good. It works fine in my basic testing. I'll merge it in. Thanks!
